R/license_note.R

```diff
@@ -37,23 +30,36 @@ write_license_note <- function(path = ".", force = TRUE) {
     )
   }
 
+  cargo_metadata_common_args <- c(
+    "--format-version", "1",
+    "--manifest-path", manifest_file
+  )
+
+  package_names <- processx::run(
+    "cargo",
+    c(
+      "metadata",
+      cargo_metadata_common_args,
+      "--no-deps"
+    )
+  )$stdout %>%
+    jsonlite::parse_json() %>%
+    purrr::pluck("packages") %>%
+    purrr::map_chr("name")
+
   list_license <- processx::run(
     "cargo",
     c(
-      "license",
-      "--authors",
-      "--json",
-      "--avoid-build-deps",
-      "--avoid-dev-deps",
-      "--manifest-path", manifest_file
+      "metadata",
+      cargo_metadata_common_args
     )
   )$stdout %>%
-    jsonlite::parse_json()
+    jsonlite::parse_json() %>%
+    purrr::pluck("packages")
 
   .prep_authors <- function(authors, package) {
-    ifelse(!is.null(authors), authors, paste0(package, " authors")) %>%
-      stringi::stri_replace_all("", regex = r"(\ <.+?>)") %>%
-      stringi::stri_replace_all(", ", regex = r"(\|)")
+    ifelse(!rlang::is_empty(authors), authors, paste0(package, " authors")) %>%
+      stringi::stri_replace_all("", regex = r"(\ <.+?>)")
```

Comment:
I think the standard parameter order is (string, detected pattern, replacement). There should be an `stri_replace_all_regex()` version to simplify this.